src/kvstore/wal/FileBasedWal.cpp

```diff
@@ -48,9 +44,17 @@ FileBasedWal::FileBasedWal(const folly::StringPiece dir,
 
     scanAllWalFiles();
     if (!walFiles_.empty()) {
+        firstLogId_ = walFiles_.begin()->second->firstId();
         auto& info = walFiles_.rbegin()->second;
-        lastLogId_ = info->lastId();
-        lastLogTerm_ = info->lastTerm();
+        if (info->lastId() <= 0 && walFiles_.rbegin()++ != walFiles_.rend()) {
```

Comment:
what is the condition mean "walFiles_.rbegin()++ != walFiles_.rend()"? It look like useless about ++.

Reply:
Good catch!! 👍  Shame on myself. Supposed to be
```
++walFiles_.rbegin() != walFiles_.rend()
```